src/stdlib.ts

```diff
@@ -1,17 +1,1788 @@
+        handleRuntimeError(context, new MissingRequiredPositionalError(source, (command as es.Node), 'isinstance', Number(2), args, false));
+    } else if (args.length > 2) {
+        handleRuntimeError(context, new TooManyPositionalArgumentsError(source, (command as es.Node), 'isinstance', Number(2), args, false));
+    }
+  
+    const obj = args[0];
+    const classinfo = args[1];
+
+    let expectedType: string;
+    if (classinfo.type === 'string') {
+        switch (classinfo.value) {
+            case 'int':
+                expectedType = 'bigint';
+                break;
+            case 'float':
+                expectedType = 'number';
+                break;
+            case 'string':
+                expectedType = 'string';
+                break;
+            case 'bool':
+                expectedType = 'bool';
+                break;
+            case 'complex':
+                expectedType = 'complex';
+                break;
+            case 'NoneType':
+                expectedType = 'NoneType';
+                break;
+            default:
+                handleRuntimeError(context, new ValueError(source, command as es.Node, context, "isinstance"));
+                return;
+        }
+    } else {
+        handleRuntimeError(context, new TypeError(source, command as es.Node, context, args[0].type, "string"));
+        return;
+    }
+
+    const result = obj.type === expectedType;
+  
+    return { type: 'bool', value: result };
+}
+
+export function math_acos(args: Value[], source: string, command: ControlItem, context: Context): Value {
+    if (args.length < 1) {
+        handleRuntimeError(context, new MissingRequiredPositionalError(source, (command as es.Node), 'math_acos', Number(1), args, false));
+    } else if (args.length > 1) {
+        handleRuntimeError(context, new TooManyPositionalArgumentsError(source, (command as es.Node), 'math_acos', Number(1), args, false));
+    }
+  
+    const x = args[0];
+    if (x.type !== 'number' && x.type !== 'bigint') {
+        handleRuntimeError(context, new TypeError(source, command as es.Node, context, x.type, "float' or 'int"));
+    }
+
+    let num: number;
+    if (x.type === 'number') {
+        num = x.value;
+    } else {
+        num = Number(x.value);
+    }
+
+    if (num < -1 || num > 1) {
+        handleRuntimeError(context, new ValueError(source, command as es.Node, context, "math_acos"));
+    }
+  
+    const result = Math.acos(num);
+    return { type: 'number', value: result };
+}
+
+export function math_acosh(args: Value[], source: string, command: ControlItem, context: Context): Value {
+    if (args.length < 1) {
+        handleRuntimeError(context, new MissingRequiredPositionalError(source, (command as es.Node), 'math_acosh', Number(1), args, false));
+    } else if (args.length > 1) {
+        handleRuntimeError(context, new TooManyPositionalArgumentsError(source, (command as es.Node), 'math_acosh', Number(1), args, false));
+    }
```

Comment:
I noticed that for each of this, the argument parsing is mostly the same.

If so, can we add a decorator to do the argument parsing? Basically take a higher order function https://www.typescriptlang.org/docs/handbook/decorators.html.

Something like

```
@validate(1, "math_acosh") // 1 is the number of expected arguments.
export function math_acosh(...)
```